CHANGELOG.md

```diff
@@ -4,13 +4,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
-- No changes yet.
+## [1.8.0] - 2019-03-27
+### Added
+- Added `As` option for `Provide` to specify that the value produced by the
```

Comment:
```
dig.Provide(func () Foo { ... }, dig.As(new(bar), new(baz))`
```

Just inserts `bar` and `baz` into the graph, but not `Foo`, correct?

Reply:
Foo as well. Docs clarify this with example of what the code is equivalent to,

https://github.com/uber-go/dig/blob/master/dig.go#L164-L171

Reply:
I think that may be enough for me to lobby for a train halt here. I don't think `Foo` should be included since that may not be the intention at all.

`dig.As` seems to imply _only as_, rather than `dig.AlsoAs` which is the current semantic.

Reply:
I agree, the name can be confusing but the semantic is correct given that before this function existed I used to return multiple types to satisfy implemented interfaces by the original type and the name should state that intention pretty clear.

To me the following names fit that intention:

* `AlsoAs`: `c.Provide(newBuffer, dig.AlsoAs(new(io.Reader), new(io.Writer)))`
* `TypeAlias`: `c.Provide(newBuffer, dig.TypeAlias(new(io.Reader), new(io.Writer)))`
* `Satisfy`: `c.Provide(newBuffer, dig.Satisfy(new(io.Reader), new(io.Writer)))`
* `Implement`: `c.Provide(newBuffer, dig.Implement(new(io.Reader), new(io.Writer)))`